takanon_extractions/dynamic_extractions/extract_content.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python3
+"""
+Extract content for specific section types from HTML files based on structure JSON.
+"""
+
+import argparse
+import json
+import sys
+from pathlib import Path
+from bs4 import BeautifulSoup
+from markdownify import markdownify as md
+from botnim.config import get_logger
+
+# Logger setup
+logger = get_logger(__name__)
+
+def extract_content_for_sections(html_content, structure_data, target_content_type):
+    """
+    Extract complete content for sections of the specified type.
+    
+    Args:
+        html_content: Raw HTML content
+        structure_data: Parsed JSON structure
+        target_content_type: Type of content to extract (e.g., "סעיף")
+    
+    Returns:
+        Updated structure with content added
+    """
+    soup = BeautifulSoup(html_content, 'html.parser')
+    
+    # Find all sections with html_id in the structure
+    sections_with_ids = []
```

Comment:
A different approach that doesn't require a "global variable":

```python
    def collect_sections(items):
        for item in items:
            if item.get('html_id'):
                yield item
            if 'children' in item:
                yield from collect_sections(item['children'])

    sections_with_ids = list(collect_sections(structure_data['structure']))
    ```

Reply:
fixed. waiting for the day I can recognize these anti patterns myself :)